mkdocs_juvix/plugin.py

```diff
@@ -82,10 +82,26 @@ def on_page_read_source(self, page: Page, config):
         return page.content
 
     def on_page_markdown(self, markdown, page, config, files: Files):
-        if page.file.name.endswith(".juvix"):
-            page.file.name = page.file.name.replace(".juvix", "")
-            page.file.url = page.file.url.replace(".juvix", "")
-            page.file.dest_uri = page.file.dest_uri.replace(".juvix", "")
+        index = "index.juvix"
+        juvix = ".juvix"
+
+        def path_change(text):
+            page.file.name = page.file.name.replace(text, "")
+            page.file.url = page.file.url.replace(text, "")
+            page.file.dest_uri = page.file.dest_uri.replace(text, "")
             page.file.abs_dest_path = page.file.abs_dest_path.replace(
-                ".juvix", "")
+                text, "")
+
+        if page.file.name.endswith(index):
+            path_change(index)
+        elif page.file.name.endswith(juvix):
+            path_change(juvix)
         return markdown
+
+    def on_nav(self, nav: Navigation, config: Config, files: Files):
+
+        for page in nav.pages:
+           if page.file.src_path.endswith("index.juvix.md"):
```

Comment:
We need to verify that the path is within the docs folder to consider it homepage, as it could be in a different subfolder.